cmd/noble/broadcast.go

```diff
@@ -131,50 +131,74 @@ func Broadcast(
 		}
 
 		rpcResponse, err := rpcClient.BroadcastTxSync(context.Background(), txBytes)
-		if err == nil && rpcResponse.Code == 0 {
-			msg.Status = types.Complete
-			return rpcResponse, nil
-		}
+		if err != nil || (rpcResponse != nil && rpcResponse.Code != 0) {
+			// Log the error
+			logger.Error(fmt.Sprintf("error during broadcast: %s", getErrorString(err, rpcResponse)))
+
+			if err != nil || rpcResponse == nil {
```

Comment:
Since this is nested in the previous "if" statement, is this `if err != ni` necessary?